spec/bookshelf-manager-v1.yaml

```diff
@@ -0,0 +1,186 @@
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '404':
+          description: No books found
+    post:
+      tags:
+        - books
+      summary: Add a new book
+      description: Add a new book
+      security:
+        - ApiKeyHeader: []
+        - ApiKeyQuery: []
+      requestBody:
+        description: Create a new book
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Book'
+        required: true
+      responses:
+        '201':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '400':
+          description: Invalid book ID supplied
+    delete:
+      tags:
+        - admin
+      summary: Delete all books
+      description: Delete all books
+      security:
+        - OAuth2: [admin]
+          ApiKeyHeader: []
+        - OAuth2: [admin]
+          ApiKeyQuery: []
+      responses:
+        '204':
+          description: Successful operation
+        '401':
+          description: Not authenticated
+        '403':
+          description: Access token does not have the required scope
+  /books/{bookId}:
+    get:
+      tags:
+        - books
+      summary: Find book by ID
+      description: Returns a single book
+      parameters:
+        - name: bookId
+          in: path
+          description: ID of book to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Successful retrieval of the book
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '400':
+          description: Invalid book ID supplied
+        '404':
+          description: Book not found
+    put:
+      tags:
+        - books
+      summary: Update an existing book by ID
+      description: Update an existing book by ID
+      security:
+        - ApiKeyHeader: []
+        - ApiKeyQuery: []
+      parameters:
+        - name: bookId
+          in: path
+          description: ID of book to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '400':
+          description: Invalid book ID supplied
+        '404':
+          description: Book not found
+    delete:
+      tags:
+        - admin
+      summary: Delete a book by ID
+      description: Delete a book by ID
+      security:
+        - OAuth2: [admin]
+          ApiKeyHeader: []
+        - OAuth2: [admin]
+          ApiKeyQuery: []
```

Comment:
DELETE endpoints should require an Auth Token **AND** API key (via header or query) for robustness